.github/workflows/cli-test.yml

```diff
@@ -0,0 +1,55 @@
+name: Cli Tests
+on:
+  push:
+     branches: [ main ]
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [16.x, 14.x, 10.x]
+    steps:
+      - name: Checkout cli repo
+        uses: actions/checkout@v2
+      - run: npm install
```

Comment:
From what I understand, this would just test out the `branch` along with the PR commit?

Wouldn't we need a `merge` test as well? eg: Consider having a branch (imagine `release-feature-branch` and which diverted ahead from main with a lot more changes). Now let's say we'd need to merge `release-feature-branch` to main via a PR. From what I get these steps do the following: 

- Checkout `release-feature-branch`
- Add the PR commit on top of it
- Run npm tests

What we'll missing on is the post-merge behaviour on the `main` branch. There are chances of builds failing post a merger, even if the branch builds succeed. I remember the pipeline having `Pull Request Builds` earlier with travis which exactly does the same, in addition with `branch builds`. Thoughts?

Reply:
The default ref of [actions/checkout@v2](https://github.com/actions/checkout#:~:text=the%20default%20branch.-,ref,-%3A%20%27%27%0A%0A%20%20%20%20%23%20Personal%20access%20token) is to the reference or SHA for that event. The GITHUB_REF of pull request event is the [PR merge branch](https://docs.github.com/en/enterprise-server@2.22/actions/learn-github-actions/events-that-trigger-workflows#:~:text=PR%20merge%20branch-,refs/pull/%3AprNumber/merge,-You%20extend%20or). So if we doesn’t specify with ref part, it checks out the merged source code. 

If we see on our [PR](https://github.com/twilio/twilio-cli/pull/312/checks?check_run_id=3948082739#step:2:380), the build is running on [merge commit](https://github.com/twilio/twilio-cli/commit/91eaa5968c67a1aee92bab2cceafe734ffb111d3) not on the latest commit of  current PR. I think that's why when there is a merge conflict with the base branch it [will not trigger the PR builds](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#:~:text=Workflows%20will%20not%20run%20on%20pull_request%20activity%20if%20the%20pull%20request%20has%20a%20merge%20conflict.%20The%20merge%20conflict%20must%20be%20resolved%20first.). @onuzbee Please let me know If I am missing something here.